src/CMakeLists.txt

```diff
@@ -255,19 +255,26 @@ ELSE()
 ENDIF()
 
 # Optional external dependency: zlib
-FIND_PACKAGE(ZLIB)
-IF (ZLIB_FOUND)
-	LIST(APPEND LIBGIT2_INCLUDES ${ZLIB_INCLUDE_DIRS})
-	LIST(APPEND LIBGIT2_LIBS ${ZLIB_LIBRARIES})
-	IF(APPLE OR CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
-		LIST(APPEND LIBGIT2_LIBS "z")
-		LIST(APPEND LIBGIT2_PC_LIBS "-lz")
+IF (NOT USE_BUNDLED_ZLIB)
+	FIND_PACKAGE(ZLIB)
+	IF (NOT USE_BUNDLED_ZLIP AND ZLIB_FOUND)
```

Comment:
`NOT USE_BUNDLED_ZLIB` is not required here, as it's already a prerequisite to getting here.